pims/base_frames.py

```diff
@@ -133,45 +134,163 @@ def __repr__(self):
                                   dtype=self.pixel_type)
 
 
-class FramesSequence(FramesStream):
-    """Baseclass for wrapping data buckets that have random access.
+class SliceableIterable(object):
+
+    def __init__(self, ancestor, indices, length=None):
+        """A generator that supports fancy indexing
+
+        When sliced using any iterable with a known length, it return another
+        object like itself, a SliceableIterable. When sliced with an integer,
+        it returns the data payload.
+
+        Also, this retains the attributes of the "ancestor" object that
```

Comment:
This is not quite true, you keep passing the base ancestor through, not it's immediate parent.